pkg/repository/v1_repository.go

```diff
@@ -281,6 +266,22 @@ func (r *V1Repository) updateLocalSnapshot() (*v1manifest.Snapshot, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// Update root if needed and restart the update process
+		var oldRoot v1manifest.Root
+		_, _, err = r.local.LoadManifest(&oldRoot)
+		if err != nil {
+			return nil, err
+		}
+		newRootVersion := snapshot.Meta[v1manifest.ManifestURLRoot].Version
```

Comment:
What if the latest `snapshot.json` is signed with a new key only available in the (remote) new `root.json`?

Reply:
fixed as new commit, it will update root.json when SignatureError or ExpirationError